x/valset/keeper/keeper.go

```diff
@@ -516,12 +527,19 @@ func (k Keeper) GetSigningKey(ctx sdk.Context, valAddr sdk.ValAddress, chainType
 }
 
 // IsJailed returns if the current validator is jailed or not.
-func (k Keeper) IsJailed(ctx sdk.Context, val sdk.ValAddress) bool {
-	return k.staking.Validator(ctx, val).IsJailed()
+func (k Keeper) IsJailed(ctx context.Context, val sdk.ValAddress) bool {
+	a, err := k.staking.Validator(ctx, val)
+	if err != nil {
+		panic(err)
```

Comment:
Don't panic in state machine code. Return an error if possible.

Reply:
Ok,I'll try to return the error if possible